### PR TITLE
[dashboard] Add support for the new Dashboard API

### PR DESCRIPTION
**Pull request by MLaureB**
This PR adds support for the [new Dashboard API](https://docs.datadoghq.com/api/?lang=python#dashboards) in the library.

### CODE REVIEW

**Comment by bkabrda** [comment]
:+1: LGTM. We'll have to figure out how to properly obsolete and remove things connected to the old APIs, but that's not urgent and it's better done separately anyway.

File: boards.go [hunk 1]
```diff
@@ -0,0 +1,59 @@
+/*
+ * Datadog API for Go
+ *
+ * Please see the included LICENSE file for licensing information.
+ *
+ * Copyright 2013 by authors and contributors.
```
Comment (ojongerius) [inline boards.go]: Could do with an update to 2019

File: board_widgets.go [hunk 2]
```diff
@@ -0,0 +1,712 @@
+package datadog
```
Comment (ojongerius) [inline board_widgets.go]: I know we've been a little inconsistent with it, but could you please add license information like you did for `board.go`?

File: boards.go (hunk 1 above)
Comment (bkabrda) [reply boards.go]: Updated, resolving

File: board_widgets.go (hunk 2 above)
Comment (bkabrda) [reply board_widgets.go]: Added, resolving.

**Comment by bkabrda** [comment]
@ojongerius all your comments have been addressed, so I'm going to hope that you won't mind me merging before you get to re-review.

Thanks @MLaureB!

**Pull request merged**